connector.go

```diff
@@ -92,6 +92,9 @@ type SchemaStatus struct {
 
 // Connector is the interface that must be implemented for a backend service
 // It can also be implemented using an RPC such as thrift (dosa-idl)
+// When fields are returned from read/range/search/scan methods, it's legal for the connector
+// to return more fields than originally requested. The caller of the connector should never mutate
+// the returned columns either, in case they are from a cache
```

Comment:
This feels really brittle to me. I can't quite articulate why though, so I won't block.

That said, if we're really going to make this change to the Connector contract, I would advise we change the name of the `fieldsToRead` parameter in read/range/search/scan functions. It's not obvious in those functions that parameter `fieldToRead` actually means "please get at least these fields but it's also cool if you also give me additional fields". Perhaps rename them to `requiredFields`?